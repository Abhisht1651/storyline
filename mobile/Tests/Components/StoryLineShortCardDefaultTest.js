// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import StoryLineShortCardDefault from '../../App/Components/StoryLineShortCardDefualt'

describe('StoryLineShortCardDefault', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineShortCardDefault />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineShortCardDefault />)
    expect(wrapper.find('Text')).toHaveLength(6)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineShortCardDefault />)
    expect(wrapper.find('Image')).toHaveLength(4)
  })
})
