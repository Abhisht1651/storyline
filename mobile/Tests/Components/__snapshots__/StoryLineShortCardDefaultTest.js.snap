// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryLineShortCardDefault should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 30,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": "100%",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 145,
        "paddingBottom": 34,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": "8.0745%",
          "width": "87.3%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.08,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          World
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#1F1F60",
              "borderRadius": 2,
              "height": 4,
              "marginHorizontal": 10,
              "width": 4,
            }
          }
        />
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "marginRight": 6,
              "width": 34,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.08,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Trending
        </Text>
      </View>
      <ForwardRef>
        <ForwardRef>
          <X4SButton
            disabled={true}
            onPress={[Function]}
            outline={false}
            style={Object {}}
            title="Following"
          />
        </ForwardRef>
      </ForwardRef>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": "0.8%",
          "width": "87.3%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 26,
              "width": 26,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "opacity": 1,
              "paddingRight": 40,
              "textAlign": "left",
            }
          }
        >
           
          45
          K Reached
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 26,
              "width": 26,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "opacity": 1,
              "paddingRight": 40,
              "textAlign": "left",
            }
          }
        >
           
          45 friends Following
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": "7.4534%",
          "width": "87.3%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 32,
            "letterSpacing": -0.11,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        UK exit from the European Union
      </Text>
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 24,
            "letterSpacing": -0.07,
            "marginTop": "1.5527%",
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        Updated 
        2m
         ago
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "default": "",
      }
    }
    style={
      Object {
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "flex": 177,
        "opacity": 1,
        "width": "100%",
      }
    }
  />
</View>
`;
